### ISSUE TITLE
eBPF plugin doesnt start after #9234 has been merged

### ISSUE THREAD

**Issue by vlvkobal** [netdata/netdata#9567]
##### Bug report summary
eBPF plugin can't start after `netdata.service` configuration was changed in #9234
```
2020-07-20 16:02:52: netdata INFO  : PLUGINSD[ebpf] : connected to '/usr/libexec/netdata/plugins.d/ebpf.plugin' running on pid 133723
2020-07-20 16:02:52: ebpf.plugin ERROR : MAIN : Setrlimit(RLIMIT_MEMLOCK) (errno 1, Operation not permitted)
2020-07-20 16:02:52: netdata ERROR : PLUGINSD[ebpf] : read failed: end of file (errno 9, Bad file descriptor)
2020-07-20 16:02:52: netdata INFO  : PLUGINSD[ebpf] : PARSER ended
2020-07-20 16:02:52: netdata ERROR : PLUGINSD[ebpf] : '/usr/libexec/netdata/plugins.d/ebpf.plugin' (pid 133723) disconnected after 0 successful data collections (ENDs).
2020-07-20 16:02:52: netdata ERROR : PLUGINSD[ebpf] : child pid 133723 exited with code 4.
2020-07-20 16:02:52: netdata ERROR : PLUGINSD[ebpf] : '/usr/libexec/netdata/plugins.d/ebpf.plugin' (pid 133723) exited with error code 4 and haven't collected any data. Disabling it.
2020-07-20 16:02:52: netdata INFO  : PLUGINSD[ebpf] : thread with task id 133712 finished
```

##### OS / Environment
Arch Linux - 5.7.9

##### Netdata version
netdata v1.23.2-28-geb42d2ad4

##### Component Name
systemd configuration

##### Steps To Reproduce
Restart Netdata, check if ebpf.plugin is running.

##### Expected behavior
ebpf.plugin doesn't crash on systems with systemd.

**Comment by thiagoftsm** [comment]
@vlvkobal only root can do operations with  `Setrlimit(RLIMIT_MEMLOCK)` , it looks like that permission was missed.

**Comment by Ferroin** [comment]
> @vlvkobal only root can do operations with `Setrlimit(RLIMIT_MEMLOCK)` , it looks like that permission was missed.

After looking a bit further, I believe this is indeed the issue.

However, I'm curious why the eBPF plugin needs to manipulate resource limits like this. Is it in some way required for correct functioning of the plugin (more specifically in this case, do we actually _need_ to lock memory to get the plugin to work correctly, as I assume we're raising this limit so we can use `memlock()`)?

**Comment by thiagoftsm** [comment]
@Ferroin we set this, because this plugin is growing and we will have more memory management coming in the next months, no less important this is a common call among kernel [examples](https://github.com/torvalds/linux/tree/master/samples/bpf), mainly the socket examples.

**Comment by Ferroin** [comment]
@thiagoftsm My point though is that we probably don't actually _need_ this and should gracefully handle if setting rlimigs does not work and we can't lock memory. Ideally this should be behind a config option, because there are perfectly legitimate reasons for user to _not_ want us to lock memory.

**Comment by Ferroin** [comment]
OK, the issue is indeed the call to `setrlimit()` and the lack of `CAP_SYS_RESOURCE` (and the failure to successfully lock memory resulting from that). We _really_ should not require this to work, and I'd argue shouldn't even be trying to do it if we can't substantiate _why_ we think we need it. Locking memory has the potential to cause all kinds of issues elsewhere on the system, and you should always be able to explain in detail _exactly_ why you are doing it.

**Comment by Mic92** [comment]
Fixed by https://github.com/netdata/netdata/pull/9569 ?

**Comment by prologic** [comment]
> Fixed by #9569 ?

Possibly. Can you confirm?

**Comment by Mic92** [comment]
> > Fixed by #9569 ?
> 
> Possibly. Can you confirm?

Sorry, cannot test right now. We have not packaged that for NixOS. But maybe the issue author can have a look.

**Comment by ilyam8** [comment]
> eBPF plugin can't start
> Arch Linux - 5.7.9

I am on manjaro (5.7.9), eBPF plugin works on latest, i did clean install

**Comment by prologic** [comment]
@ilyam8 Do you think we can close this issue then?

**Comment by ilyam8** [comment]
i think yes, eBPF works for me on latest

i get these warning during compilation but that is another issue

```cmd
In function 'fill_resolved_name',
    inlined from 'fill_names.part.0' at collectors/ebpf.plugin/ebpf_socket.c:656:5:
collectors/ebpf.plugin/ebpf_socket.c:546:9: warning: 'memcpy' writing 1018 bytes into a region of size 980 [-Wstringop-overflow=]
  546 |         memcpy(ptr->resolved_name, hostname, length);
      |         ^
collectors/ebpf.plugin/ebpf_socket.c: In function 'fill_names.part.0':
libnetdata/libnetdata.c:177:15: note: at offset 0 to an object with size 980 allocated by 'malloc' here
  177 |     void *p = malloc(size);
      |               ^
In function 'fill_resolved_name',
    inlined from 'fill_names.part.0' at collectors/ebpf.plugin/ebpf_socket.c:656:5:
collectors/ebpf.plugin/ebpf_socket.c:547:36: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
  547 |         ptr->resolved_name[length] = '\0';
      |                                    ^
collectors/ebpf.plugin/ebpf_socket.c: In function 'fill_names.part.0':
libnetdata/libnetdata.c:177:15: note: at offset 1018 to an object with size 0 allocated by 'malloc' here
  177 |     void *p = malloc(size);
      |
```
